src/mem/epoch/markable_atomic.rs

```diff
@@ -48,32 +53,53 @@ fn opt_owned_into_usize<T>(ptr: Option<Owned<T>>, mark: usize) -> usize {
     raw as usize | mark
 }
 
-impl<T> MarkedAtomic<T> {
-    /// Create a new, null atomic pointer.
+impl<T> MarkableAtomic<T> {
+    /// Create a new, null markable atomic pointer with a marker value of 0.
     #[cfg(feature = "nightly")]
-    pub const fn null() -> MarkedAtomic<T> {
-        MarkedAtomic {
+    pub const fn zero() -> MarkableAtomic<T> {
+        MarkableAtomic {
             ptr: atomic::AtomicUsize::new(0),
-            _marker: PhantomData
+            _marker: PhantomData,
         }
     }
 
+    /// Create a new, null markable atomic pointer with a marker value of 0.
     #[cfg(not(feature = "nightly"))]
-    pub fn null() -> MarkedAtomic<T> {
-        MarkedAtomic {
+    pub fn zero() -> MarkableAtomic<T> {
+        MarkableAtomic {
             ptr: atomic::AtomicUsize::new(0),
-            _marker: PhantomData
+            _marker: PhantomData,
+        }
+    }
+
+    /// Create a new, null markable atomic pointer with the given marker value.
+    pub fn null(mark: usize) -> MarkableAtomic<T> {
+        guard_mark::<T>(mark);
+        MarkableAtomic {
+            ptr: atomic::AtomicUsize::new(mark),
+            _marker: PhantomData,
         }
     }
 
-    /// Create a new atomic pointer
-    pub fn new(data: T, mark: usize) -> MarkedAtomic<T> {
+    /// Create a new markable atomic pointer.
+    pub fn new(data: T, mark: usize) -> MarkableAtomic<T> {
         guard_mark::<T>(mark);
-        MarkedAtomic {
+        MarkableAtomic {
```

Comment:
Obviously this code is just what was there previously, but I think a call to `from_raw` would be better here so as to not duplicate code.

That said, it is probably good to call `guard_mark` before allocating a box... so I'm sort of conflicted.